code/game/machinery/bees_apiary.dm

```diff
@@ -73,53 +76,78 @@ var/list/apiaries_list = list()
 			B.mob.home = null
 	..()
 
+/obj/machinery/apiary/wrenchAnchor(var/mob/user, var/obj/item/I, var/time_to_wrench = 3 SECONDS)
+	. = ..()
+	if (.)
+		update_icon()
+
+/obj/machinery/apiary/attack_hand(var/mob/user)
+	if(reagents.total_volume <= 0)
+		alert(user,"There's no honey to harvest yet!","[name]","Ok")
+		return
+
+	if(alert(user,"Harvest the honeycombs?[((queen_bees_inside || worker_bees_inside) && species.angery) ? " Be ready to handle some angry bees!" : ""]","[name]","Yes","No")== "Yes")
+		user.visible_message("<span class='notice'>\the [user] begins dismantling the apiary.</span>","<span class='danger'>You begin harvesting the honeycombs.</span>")
+
+		if((queen_bees_inside || worker_bees_inside) && species.angery)
+			user.visible_message("<span class='danger'>The [species.common_name] don't like that.</span>")
+			angry_swarm(user)
+
+		if(do_after(user, loc, 50))
+			if(harvest_honeycombs())
+				to_chat(user, "<span class='notice'>You successfully harvest the honeycombs.</span>")
+			else
+				to_chat(user, "<span class='notice'>You somehow didn't find a single honeycomb in there.</span>")
+
 /obj/machinery/apiary/update_icon()
 	overlays.len = 0
-	overlays += image('icons/obj/apiary_bees_etc.dmi', icon_state=apiary_icon)
+	icon_state = "[apiary_icon][anchored ? "-anchored" : ""]"
 
 	var/image/I = null
 	switch(reagents.total_volume)
 		if(30 to 60)
-			I = image('icons/obj/apiary_bees_etc.dmi', icon_state="honey_1")
+			I = image('icons/obj/apiary_bees_etc.dmi', icon_state="[apiary_icon]_honey_1")
 		if(60 to 90)
-			I = image('icons/obj/apiary_bees_etc.dmi', icon_state="honey_2")
+			I = image('icons/obj/apiary_bees_etc.dmi', icon_state="[apiary_icon]_honey_2")
 		if(90 to INFINITY)
-			I = image('icons/obj/apiary_bees_etc.dmi', icon_state="honey_3")
+			I = image('icons/obj/apiary_bees_etc.dmi', icon_state="[apiary_icon]_honey_3")
 	if(!I)
 		return
 	I.color = mix_color_from_reagents(reagents.reagent_list)
 	overlays += I
+	if (on_fire)
+		overlays += fire_overlay
```

Comment:
Did you test to ensure this doesn't result in stacked fire overlays?

Reply:
The first line of the proc removes every overlays. So no but I did test burning hives before adding that and the fire overlay went away any time udpate_icon()  was called, so this actually fixes that.